blackbox/docs/best_practice/sharding.txt

```diff
@@ -0,0 +1,97 @@
+.. highlight:: sh
+
+.. _sharding_guide:
+
+==============
+Sharding Guide
+==============
+
+This document is a sharding best practice guide for CrateDB.
+
+A brief recap: CrateDB tables are split into a configured number of shards, and
+then these shards are distributed across the cluster.
+
+Figuring out how many shards to use for your tables requires you to think about
+the type of data you're processing, the types of queries you're running, and
+the type of hardware you're using.
+
+.. NOTE::
+
+   This guide assumes you know the basics. If you are looking for an intro to
+   sharding, see :ref:`sql_ddl_sharding`.
+
+Optimising For Query Performance
+================================
+
+Under-Allocation Is Bad
+-----------------------
+
+.. CAUTION::
+
+   If you have fewer shards than CPUs in the cluster, this is called
+   *under-allocation*, and it means you're not getting the best performance out
+   of CrateDB.
+
+Whenever possible, CrateDB will parallelize query workloads and distribute them
+across the whole cluster. The more CPUs this query workload can be distributed
+across, the faster the query will run.
+
+To increase the chances that a query can be parallelized and distributed
+maximally, there should be at least as many shards for a table than there are
+CPUs in the cluster. This is because CrateDB will automatically balance shards
+across the cluster so that each node contains as few shards as possible.
+
+In summary: the smaller your shards are, the more of them you will have, and so
+the more likely it is that they will be distributed across the whole cluster,
+and hence across all of your CPUs, and hence the faster your queries will run.
+
+Significant Over-Allocation Is Bad
+----------------------------------
+
+.. CAUTION::
+
+   If you have more shards than CPUs, this is called *over-allocation*. A
+   little over-allocation is desirable. But if you significantly over-allocate
+   your shards, you will see performance degradation.
+
+When you have slightly more shards per table than CPUs, you ensure that query
+workloads can be parallelized and distributed maximally, which in turn ensures
+maximal query performance.
+
+However, if most nodes hold have more shards per table than they have CPUs, you
+could actually see performance degradation. Each shard comes with a cost in
+terms of open files, RAM, and CPU cycles. Smaller shards also means small shard
+indexes, which can adversely affect computed search term relevance.
+
```

Comment:
In CrateDb we don't recommend to have more than 1k shards per node and index. The user will get a cluster check warning if this is the case. Maybe it's worth to mention this.